sda-commons-server-dropwizard/src/main/java/org/sdase/commons/server/dropwizard/bundles/configuration/generic/GenericLookupYamlConfigurationFactory.java

```diff
@@ -62,16 +70,100 @@ protected T build(JsonNode node, String path) throws IOException, ConfigurationE
     mappableFields.stream()
         .map(m -> m.expand(configurationRuntimeContext.getDefinedKeys()))
         .flatMap(List::stream)
+        .sorted()
         .forEach(mappableField -> configure(node, mappableField));
     return super.build(node, path);
   }
 
   private void configure(JsonNode baseNode, MappableField mappableField) {
-    String jsonPathOfField = String.join(".", mappableField.getJsonPathToProperty());
-    addOverride(
-        baseNode,
-        jsonPathOfField,
-        configurationRuntimeContext.getValue(mappableField.getContextKey()));
+    String value = configurationRuntimeContext.getValue(mappableField.getContextKey());
+    try {
+      String jsonPathOfField = String.join(".", mappableField.getJsonPathToProperty());
+      addOverride(baseNode, jsonPathOfField, value);
+    } catch (IllegalArgumentException e) {
+      LOG.info(
+          "{} is not mappable with Dropwizard overrides, trying best guess approach",
+          mappableField,
+          e);
+      addCustomOverride(baseNode, mappableField.getJsonPathToProperty(), value);
+    }
+  }
+
+  private void addCustomOverride(JsonNode rootNode, List<String> path, String value) {
+    try {
+      if (path.isEmpty()) {
+        return;
+      }
+      var propertyName = path.get(0);
+      var valueNow = path.size() == 1;
+
+      // setting values
+      if (valueNow) {
+        addValue(rootNode, value, propertyName);
+        return;
+      }
+
+      // building the hierarchy
+      JsonNode nextNode = addNextNodeInHierarchy(rootNode, path, propertyName);
+      addCustomOverride(nextNode, path.subList(1, path.size()), value);
+
+    } catch (Exception e) {
+      if (e instanceof IllegalArgumentException) {
+        throw e;
+      }
+      LOG.error("Failed to create an override for {}", path, e);
+    }
```

Comment:
I wondered about the `IllegalArgumentException`and double checked the code. For me it was not clear, that we throw the `IllegalArgumentException`s ourselves. My first impression was that we have to take care of some complicated reflection / magic stuff.

What about using our own custom exception here, e.g. `DynamicConfigurationException`? We could search for it and quickly find out that we don't expect exceptions from underlying Java core classes or Jackson.

Reply:
I choose the same exception that is thrown by the Dropwizard override implementation, so that any error handling in the Dropwizard code that calls this function behaves the same.